### PR TITLE
GLEW and qwt rebuild

### PR DESCRIPTION
**Pull request by ig-or**

### CODE REVIEW

**Comment by ig-or** [comment]
Looks like we cannot simply rebuild them now..  
`Project ERROR: Cannot run target compiler '/opt/bin/x86_64-w64-mingw32-libgfortran5-cxx11/x86_64-w64-mingw32-g++'. `


```
[01:01:04] install -d -m 0755 "/workspace/destdir/include/GL"
[01:01:04] install -m 0644 include/GL/wglew.h "/workspace/destdir/include/GL/"
[01:01:04] install -m 0644 include/GL/glew.h "/workspace/destdir/include/GL/"
[01:01:04] install -m 0644 include/GL/glxew.h "/workspace/destdir/include/GL/"
[01:01:04] cc -DGLEW_NO_GLU -DGLEW_BUILD -O2 -Wall -W -Iinclude -I/workspace/destdir/include -fPIC -Wcast-qual -ansi -pedantic -fno-stack-protector  -o tmp/linux/default/shared/glew.o -c src/glew.c
[01:01:04] In file included from include/GL/glxew.h:98:0,
[01:01:04]                  from src/glew.c:55:
[01:01:04] /opt/x86_64-linux-gnu/bin/../x86_64-linux-gnu/sys-root/usr/local/include/X11/Xlib.h:44:19: fatal error: X11/X.h: No such file or directory
[01:01:04]  #include <X11/X.h>
[01:01:04]                    ^
[01:01:04] compilation terminated.
```

**Comment by giordano** [comment]
@barche I think I need your help here.  qwt uses `qmake`, but when building for `x86_64-w64-mingw32` compilation fails because it tries to call `/opt/bin/x86_64-w64-mingw32-libgfortran5-cxx11/x86_64-w64-mingw32-g++`, which doesn't exist, the existing compiler is `/opt/bin/x86_64-w64-mingw32-libgfortran3-cxx11/x86_64-w64-mingw32-g++` (note the different libgfortran version in the name of the directory).  `qmake` is a binary executable, which doesn't exactly make it easy to debug the issue.  Do you know where this path is coming from?  It'd seem it's hardcoded in the binary blob (otherwise how on Earth it'd try to call a program it doesn't exist at all), if so, that sounds like an awful idea.

**Comment by barche** [comment]
Not sure, trying again with Qt5Base_jll as dep. This is the newer one that should contain proper cross-compiling qmake.

File: Q/qwt/build_tarballs.jl [hunk 1]
```diff
@@ -33,11 +34,21 @@ products = [
...
+
+if any(should_build_platform.(triplet.(platforms_linux)))
+    build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"7")
+end
+if any(should_build_platform.(triplet.(platforms_win)))
+    build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"8")
+end
+if any(should_build_platform.(triplet.(platforms_macos)))
+    build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"7")
+end
```
Comment (giordano) [inline Q/qwt/build_tarballs.jl:54]: This looks unnecessary?  This package is only for x86_64 Linux and Windows

Comment (barche) [reply Q/qwt/build_tarballs.jl]: Yes, I removed the macos one.

**Comment by barche** [comment]
It builds now, but I did indeed have to bump the GCC version. I think there is no way around that, since these are the versions used to build Qt5 (and the minimum version for that).

**Comment by giordano** [comment]
Ok, well, the dance of moving `qmake` around was uselss, the only substantial change was to use GCC 8 for Windows because apparently `qmake` hardcodes that path, as I initially suggested.  This is incredibly unfortunate, it forces us to *always* use the same version of libgfortran/C++ string ABI.

I also restored `Qt_jll`: this is only a rebuild of the package, we aren't changing the version number, changing dependencies would be disruptive.

**Pull request merged**